gbdk-lib/examples/cross-platform/platformer_template/src/player.c

```diff
@@ -0,0 +1,359 @@
+
+}
+
+void DrawPlayer(uint16_t playerRealX, uint16_t playerRealY, uint8_t frame) NONBANKED{
+    uint8_t _previous_bank = CURRENT_BANK;
+
+
+    // Get the player's position relative to the camera's position
+    uint16_t playerCameraX = (playerRealX-camera_x)+DEVICE_SPRITE_PX_OFFSET_X;
+    uint16_t playerCameraY= (playerRealY)+DEVICE_SPRITE_PX_OFFSET_Y;
+
+    // Flip horizontally, if we aren't facing right
+    
+    if(facingRight){
+
+        SWITCH_ROM(BANK(PlayerCharacterRight));
+        move_metasprite_ex(PlayerCharacterRight_metasprites[frame],0,baseProp,0,playerCameraX,playerCameraY);
+
+   } else {
+
+        SWITCH_ROM(BANK(PlayerCharacterLeft));
+        move_metasprite_ex(PlayerCharacterLeft_metasprites[frame],0,baseProp,0,playerCameraX,playerCameraY);
+   }
+
+    SWITCH_ROM(_previous_bank);
+}
+
+void UpdatePlayer(void) BANKED{
+    
+    // Use the run velocity if the B button is held
+    // Animate the threeFrameCounter faster when B is held
+    int16_t moveSpeed = (joypadCurrent & J_B) ?PLAYER_CHARACTER_RUN_VELOCITY:PLAYER_CHARACTER_WALK_VELOCITY;
+    uint8_t threeFrameCounterSpeed = (joypadCurrent & J_B) ? PLAYER_CHARACTER_RUN_TWO_FRAME_COUNTER : PLAYER_CHARACTER_WALK_TWO_FRAME_COUNTER;
+
+    threeFrameCounter+=threeFrameCounterSpeed;
+    uint8_t threeFrameCounterValue = threeFrameCounter>>4;
+    if(threeFrameCounterValue>=3){
+        threeFrameCounter=0;
+        threeFrameCounterValue=0;
+    }
+
+    uint8_t turning = FALSE;
+
+    if(joypadCurrent &J_RIGHT){
+
+        // If we are facing the other direction?
+        if(playerXVelocity<0){
+
+            // Just decrease the velocity, and save that we are turning
+            playerXVelocity+=GROUND_FRICTION;
+
+            // We are turning, IF we didn't just finish changing direction.
+            if(playerXVelocity<0)turning=TRUE;
+        }else{
+            playerXVelocity=moveSpeed;
+
+            // Switch our vram data for the player
+            if(!facingRight){
+                facingRight=TRUE;
+                UpdatePlayerVRAMTiles();
+            }
+            
+        }
+    }else if(joypadCurrent &J_LEFT){
+
+        // If we are facing the other direction?
+        if(playerXVelocity>0){
+
+            // Just decrease the velocity, and save that we are turning
+            playerXVelocity-=GROUND_FRICTION;
+
+            // We are turning, IF we didn't just finish changing direction.
+            if(playerXVelocity>0)turning=TRUE;
+        }else{
+            playerXVelocity=-moveSpeed;
+
+            // Switch our vram data for the player
+            if(facingRight){
+                facingRight=FALSE;
+                UpdatePlayerVRAMTiles();
+            }
+            
+        }
+        
+    }else{
+
+        // Move the x velocity towards 0
+        if(playerXVelocity>=GROUND_FRICTION)playerXVelocity-=GROUND_FRICTION;
+        else if(playerXVelocity<=GROUND_FRICTION)playerXVelocity+=GROUND_FRICTION;
+        else playerXVelocity=0;
```

Comment:
Seems like making this change resolves the never reaches zero movement issue (when using different motion settings). Might be another way to fix it too.
```
        // Move the x velocity towards 0
        if (playerXVelocity > 0) {
          if (playerXVelocity >= GROUND_FRICTION) playerXVelocity -= GROUND_FRICTION;
          else playerXVelocity=0;
        }
        else if (playerXVelocity < 0) {
            if (playerXVelocity <= GROUND_FRICTION) playerXVelocity += GROUND_FRICTION;
            else playerXVelocity=0;
        }
```